docs/advanced.rst

```diff
@@ -246,6 +248,73 @@ You can add custom transports using the attribute ``rtps.userTransports``.
+interface. This is mainly due to the following reasons:
+
+ * Large message support: Network protocols need to fragment data in order to comply with the specific protocol and
+   network stacks requirements.
+   SHM transport allows the copy of full messages where the only size limit is the machine's memory capacity.
+
+ * Reduce the number of memory copies: When sending the same message to different endpoints, SHM transport can
+   directly share the same memory buffer with all the destination endpoints.
+   Other protocols require to perform one copy of the message per endpoint.
+
+ * Less operating system overhead: Once initial setup is completed, shared memory transfers require much less system
+   calls than the other protocols.
+   Therefore there is a performance/time consume gain by using SHM.
+
+When two participants on the same machine have SHM transport enabled, all communications between them are automatically
+performed by SHM transport only.
+The rest of the enabled transports are not used between those two participants.
+
+To enable SHM transport in a Participant, you need to add the SharedMemTransportDescriptor to the
+``rtps.userTransports`` attribute (C++ code) or define a transport_descriptor of type SHM in the
+XML file (see below examples).
+
++--------------------------------------------------+
+| **C++**                                          |
++--------------------------------------------------+
+| .. literalinclude:: ../code/CodeTester.cpp       |
+|    :language: c++                                |
+|    :start-after: //CONF-SHM-TRANSPORT-SETTING    |
+|    :end-before: //!--                            |
++--------------------------------------------------+
+| **XML**                                          |
++--------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml        |
+|    :language: xml                                |
+|    :start-after: <!-->CONF-SHM-TRANSPORT-SETTING |
+|    :end-before: <!--><-->                        |
++--------------------------------------------------+
+
+SHM configuration parameters:
+
+ * ``segment_size``: The size of the shared memory segment in bytes.
+   A shared memory segment is created by each participant.
+   Participant's writers copy their messages into the segment and send a message reference to the destination readers.
+
+ * ``port_queue_capacity``: Each participant with SHM transport enabled listens on a queue (port) for incoming SHM
+   message references.
+   This parameter specifies the queue size (in messages).
+
+ * ``healthy_check_timeout_ms``: With SHM, Readers and writers use a queue to exchange messages (called Port).
+   If one of the processes involved crashes while using the port, the structure can be left inoperative.
+   For this reason, every time a port is opened, a healthy check is performed.
+   If the attached processes doesn't respond in ``healthy_check_timeout_ms`` milliseconds, the port is destroyed and
```

Comment:
```suggestion
   If the attached process doesn't respond in ``healthy_check_timeout_ms`` milliseconds, the port is destroyed and
```

Reply:
There can be may processes attached to a multicast port.

Reply:
Although it should really be "If the attached processes don't"